### PR TITLE
[Logs] Handle docker logs without header

### PR DESCRIPTION
**Pull request by mnshdw**
### What does this PR do?

This PR originally aimed to fix an issue that occurs when tailing Docker logs from a container running with --tty (-t). When this flag is present, logs are not prefixed by the usual 8 byte header, which meant we parsed the log timestamp and content incorrectly.

This also had the side effect of storing incorrect timestamps in the auditor as they were left-trimmed of 8 characters, eg:
```
    "docker:cc360078bf9119ede4db385e4bf1b305a966e63fb42dac234d95bceae16cb10c": {
      "LastUpdated": "2018-07-12T15:33:58.083797546Z",
      "Offset": "12T15:33:58.076572440Z" <-- this is wrong
    },
```
which could possibly result in tailing twice all the container logs if the agent was restarted.

Last of all, I noticed we did not all handle empty messages well, for example when hitting enter in a --interactive (-i) shell. This change now discards messages if no whitespace can be found after the timestamp, instead of logging an error.

### Motivation

Several people have reported occurrences of Logs agent reporting Docker parsing problems.

### CODE REVIEW

**Comment by tmichelet** [comment]
Good catch! Thanks a lot

File: pkg/logs/docker/docker.go [hunk 1]
```diff
@@ -36,34 +36,59 @@ func ParseMessage(msg []byte) (Message, error) {
...
-		Timestamp: timestamp,
+		Timestamp: string(msg[:idx]),
 	}, nil
 }
 
+// getStatus returns the status of the message based on the value of the
+// STREAM_TYPE byte in the header. STREAM_TYPE can be 1 for stdout and 2 for
+// stderr. If it doesn't match either of these, return an empty string.
+func getStatus(msg []byte) string {
+	switch msg[0] {
```
Comment (tmichelet) [inline pkg/logs/docker/docker.go]: is it possible for the message to start with `1` or `2` when running in `tty` mode?

File: pkg/logs/docker/docker_test.go [hunk 2]
```diff
@@ -26,6 +32,26 @@ func TestParseMessageShouldSucceedWithValidInput(t *testing.T) {
...
+func TestParseMessageShouldHandleTtyMessage(t *testing.T) {
+	msg, err := ParseMessage([]byte("2018-06-14T18:27:03.246999277Z foo"))
+	assert.Nil(t, err)
+	assert.Equal(t, "2018-06-14T18:27:03.246999277Z", msg.Timestamp)
+	assert.Equal(t, message.StatusInfo, msg.Status)
+	assert.Equal(t, []byte("foo"), msg.Content)
+}
+
+func TestParseMessageShouldHandleEmptyTtyMessage(t *testing.T) {
+	msg, err := ParseMessage([]byte("2018-06-14T18:27:03.246999277Z"))
```
Comment (ajacquemot) [inline pkg/logs/docker/docker_test.go]: What about `2018-06-14T18:27:03.246999277Z ` ?

File: pkg/logs/docker/docker.go (hunk 1 above)
Comment (mnshdw) [reply pkg/logs/docker/docker.go]: Very unlikely, they represent unprintable characters (start of heading, start of text)

File: pkg/logs/docker/docker_test.go (hunk 2 above)
Comment (mnshdw) [reply pkg/logs/docker/docker_test.go]: Not sure what you mean, this is what Docker gives us when we tail a blank line from a tty container.

**Review by ajacquemot — DISMISSED** [review]
Awesome ! 🎉

**Review by ajacquemot — APPROVED** [review]

**Pull request merged**